pkg/functionconfig/reader.go

```diff
@@ -36,16 +38,50 @@ func NewReader(parentLogger logger.Logger) (*Reader, error) {
 	}, nil
 }
 
+// Merges received and base configs.
+// Received config has precedence on Spec values, while base config has precedence on Meta values
 func (r *Reader) Read(reader io.Reader, configType string, config *Config) error {
+	var receivedConfigAsMap, baseConfigAsMap map[string]interface{}
```

Comment:
received is given by codeEntryType (?)
base is given by user (?)

We would need to refine these variables name to make it more clear